homeassistant/components/plugwise/__init__.py

```diff
@@ -1 +1,151 @@
-"""Plugwise Climate (current only Anna) component for Home Assistant."""
+"""Plugwise platform for Home Assistant Core."""
+
+import asyncio
+from datetime import timedelta
+import logging
+
+from Plugwise_Smile.Smile import Smile
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+ALL_PLATFORMS = ["climate"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Plugwise platform."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Plugwise Smiles from a config entry."""
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = Smile(
+        host=entry.data["host"], password=entry.data["password"], websession=websession
+    )
+
+    try:
+        connected = await api.connect()
+
```

Comment:
Can we provide a better error when the username/password is wrong and than avoid throwing `ConfigEntryNotReady` so Home Assistant doesn't retry forever?

Reply:
Adding InvalidAuthentication to our API module

Reply:
Improved with exception handling returning False